src/test/protocol_net_tests.cpp

```diff
@@ -0,0 +1,136 @@
+// Copyright (c) 2022 The Dogecoin Core developers
+// Distributed under the MIT software license, see the accompanying
+// file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+#include "netmessagemaker.h"
+#include "netbase.h"
+#include "protocol.h"
+#include "version.h"
+
+#include "test/test_bitcoin.h"
+
+#include <boost/test/unit_test.hpp>
+
+// This tests message bodies for the networking part of the p2p protocol
+BOOST_FIXTURE_TEST_SUITE(protocol_net_tests, TestingSetup)
+
+static CNetAddr ResolveIP(const std::string& ip)
+{
+    CNetAddr addr;
+    LookupHost(ip, addr, false);
+    return addr;
+}
+
+// impl independent sequential reader for vector<unsigned char>
+static std::vector<unsigned char> ReadVch(std::vector<unsigned char> vch, std::vector<unsigned char>::const_iterator* it, size_t num)
```

Comment:
I'd find this slightly more readable as `ReadVchars`; it looks singular to me as named now. This isn't a big deal and I couldn't find much in the existing source code one way or another to see that we have a strong standard.